### DISCUSSION TITLE
关于使用 ISelect 的问题

### DISCUSSION THREAD

**Discussion by Jess-Qiu** [dotnetcore/FreeSql#1822] [Q&A] [answered]
发现一个 ISelect 的问题，为了减少代码我定义了个ISelect。如果我执行过 AnyAsync 这个 ISelect 中的 limit 会变为 1，导致我下面再次执行该 ISelect ToList  所生成的 SQL 会有个 LIMIT 1

![image](https://github.com/dotnetcore/FreeSql/assets/22249167/7e7ad830-f97d-4048-8bce-37f66a0c0772)
![image](https://github.com/dotnetcore/FreeSql/assets/22249167/d1dcdd32-9de8-4289-b533-d535168c90df)

**Comment by 2881099** [comment, marked as answer]
明白，目前是引用的方式。

如果在同一个方法内，可以考虑语法唐，局部方法复用。

https://freesql.net/extra/iselect-depcopy.html
